### PR TITLE
Clarify reuse restriction on arguments to rd_kafka_produce()

### PR DESCRIPTION
**Pull request by andrewthad**
Resolves https://github.com/edenhill/librdkafka/issues/2843

### CODE REVIEW

File: src/rdkafka.h [hunk 1]
```diff
@@ -3920,7 +3920,12 @@ rd_kafka_consumer_group_metadata_read (
  *                               partition, either set manually or by the
  *                               configured partitioner.
  *
- *    .._F_FREE and .._F_COPY are mutually exclusive.
+ *    .._F_FREE and .._F_COPY are mutually exclusive. If neither of these are
+ *    set, the caller must ensure that the memory backing \p payload is not
+ *    reused until the delivery callback is called. Other buffers passed to
```
Comment (andrewthad) [inline src/rdkafka.h:3925]: Perhaps "delivery callback is invoked" instead of "delivery callback is called"?

Comment (edenhill) [reply src/rdkafka.h:3925]: Yeah, can we also make it even more clear with something like:
"..backing \p payload remains valid and is not modified or reused until .."

File: src/rdkafka.h [hunk 2]
```diff
@@ -3920,7 +3920,12 @@ rd_kafka_consumer_group_metadata_read (
...
+ *    `rd_kafka_produce()` never have this restriction on reuse, i.e. the
```
Comment (edenhill) [inline src/rdkafka.h:3926]: never/don't

**Comment by andrewthad** [comment]
Amended the commit to address the issues discussed in the inline comments.

**Comment by edenhill** [comment]
Thank you!

**Pull request merged**